src/run_metric_on_graph.py

```diff
@@ -0,0 +1,40 @@
+"""This script runs a user-selected metric on a citation graph and saves the result"""
+
+import argparse
+import os
+import pickle
+
+import networkx as nx
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('graph_file',
+                        help='The file containing the pickled graph for a MeSH heading')
+    parser.add_argument('metric',
+                        help='The metric to calculate for the graph',
+                        choices=['betweenness_centrality', 'pagerank', 'disruption_idx'],
+                        type=str.lower)
+    parser.add_argument('out_dir',
+                        help='The dictory to store the results to')
+    args = parser.parse_args()
+
+    # Load graph
+    with open(args.graph_file, 'rb') as in_file:
+        graph = pickle.load(in_file)
+
+    # Run metric on graph
+    if args.metric == 'betweenness_centrality':
```

Comment:
I know it might be a lot of work at this point to switch to a different graph module, but I've found centrality measures to be much faster in [`graph-tool`](https://graph-tool.skewed.de/) than in `networkx`. I think `networkx` is pretty much pure Python (or at least it was last time I looked into this a few years ago) and `graph-tool` core algorithms are in C++.

Up to you if you think it's worth it - if your networks are enormous it may still be slow in `graph-tool`. IIRC betweenness centrality requires calculating all-pairs shortest paths which scales with the number of nodes squared, so if you have a lot of nodes it may be a non-starter.

Reply:
Oooh that's nice to know, thanks! I'll continue with networkx for now until I run into a bottleneck, but I may switch to graph-tool down the line once I figure out how long running everything will take.

I imagine the slowest part will be the metrics I implement, which will have to iterate over the python graph interfaces in either case 

Yeah, O(n^2) isn't great scaling in general, and definitely not when running for large n a number of times. I'll think about ways to salvage it, but the current plan is to not worry about it.